diff --git a/hipcc_wrap b/hipcc_wrap
index 9c843ace84..344e7247e9 100755
--- a/hipcc_wrap
+++ b/hipcc_wrap
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 ARGS=()
 for arg in "$@"; do
diff --git a/test/shared/GPU/ROCm/test_rocmFunctions.hip b/test/shared/GPU/ROCm/test_rocmFunctions.hip
index ddb5956950..b04368d71a 100644
--- a/test/shared/GPU/ROCm/test_rocmFunctions.hip
+++ b/test/shared/GPU/ROCm/test_rocmFunctions.hip
@@ -331,7 +331,7 @@
 #else /* HIPBLAS */
 #include "rocblas/rocblas.h"
 #ifdef WITH_AMD_ROCSOLVER
-#include <rocsolver.h>
+#include <rocsolver/rocsolver.h>
 #endif
 #endif /* HIPBLAS */
 #include "hip/hip_runtime_api.h"
