diff --git a/Makefile.in b/Makefile.in
index 312c7f2..11daefb 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -21004,7 +21004,7 @@ uninstall-man: uninstall-man1 uninstall-man3
 
 # Cuda files
 .cu.lo:
-	NVCC="$(NVCC)" libtool --mode=compile --tag=CC $(top_srcdir)/nvcc_wrap $(NVCCFLAGS) $(LDFLAGS) -I$(top_builddir)/ -I$(top_srcdir)/  -c $< -o $@
+	NVCC="$(NVCC)" libtool --mode=compile --tag=CC $(top_srcdir)/nvcc_wrap $(NVCCFLAGS) -I$(top_builddir)/ -I$(top_srcdir)/  -c $< -o $@
 define extract_interface
 	@echo "Extracting interface marked with '$1' from $@...";
 	@grep -h "^ *$1" $^ | sed 's/^ *$1//;' >> $@ || { rm $@; exit 1; }
