diff --git a/configure b/configure
index c8cb34a..3627747 100755
--- a/configure
+++ b/configure
@@ -60476,7 +60476,7 @@ if test -z "$LEX" || \
    test -n "`echo $LEX | $GREP missing`" || \
    test "`basename $LEX`" != "flex"
 then :
-  if test ! -f "$srcdir/opal/util/show_help_lex.c"
+  if test ! -f "$srcdir/opal/util/keyval/keyval_lex.c"
 then :
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: *** Could not find Flex on your system." >&5
 printf "%s\n" "$as_me: WARNING: *** Could not find Flex on your system." >&2;}
