diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a9e60fd8..e9635fa1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -119,7 +119,8 @@ if( GAUXC_HAS_MPI )
 endif()
 
 if ( GAUXC_HAS_ONEDFT )
-  target_link_libraries( gauxc PUBLIC "${TORCH_LIBRARIES}" nlohmann_json::nlohmann_json)
+  target_link_libraries( gauxc PUBLIC "${TORCH_LIBRARIES}")
+  target_link_libraries( gauxc PRIVATE nlohmann_json::nlohmann_json)
 endif()
 
 add_subdirectory( runtime_environment )
