--- a/f_check	2025-12-03 06:29:33.000000000 +0000
+++ b/f_check	2025-12-03 06:32:11.000000000 +0000
@@ -86,7 +86,11 @@
 		vendor=CRAY
 		openmp='-fopenmp'
 		;;		
-            *GNU*|*GCC*)
+            *Arm\ F90*|*F90\ Flang*|*F90\ AOCC*)
+               vendor=FLANG
+               openmp='-fopenmp'
+               ;;
+            *GNU*|*GCC*)	    
 
                 v="${data#*GCC: *\) }"
                 v="${v%%\"*}"
