diff --git a/src/dftd3/cutoff.f90 b/src/dftd3/cutoff.f90
index 743774e..8c57090 100644
--- a/src/dftd3/cutoff.f90
+++ b/src/dftd3/cutoff.f90
@@ -51,7 +51,7 @@ module dftd3_cutoff
       real(wp) :: disp3 = disp3_default

       !> Width of smooth two-body interaction cutoff
-      real(wp) :: width2 = 0.0_wp
+      real(wp) :: width2 = 0.05_wp

       !> Width of smooth three-body interaction cutoff
       real(wp) :: width3 = 0.0_wp
diff --git a/src/dftd3/damping/atm.f90 b/src/dftd3/damping/atm.f90
index 419561a..9ce6521 100644
--- a/src/dftd3/damping/atm.f90
+++ b/src/dftd3/damping/atm.f90
@@ -128,23 +128,16 @@ subroutine get_atm_dispersion_energy(mol, trans, cutoff, width, s9, rs9, alp, rv
    real(wp) :: cutoff2, c9, dE, alp3
    real(wp) :: swij, swjk, swik, dswdr, sw

-   ! Thread-private arrays for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)

    cutoff2 = cutoff*cutoff
    alp3 = alp / 3.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, trans, c6, s9, rs9, alp3, rvdw, cutoff2, cutoff, width) &
    !$omp private(iat, jat, kat, izp, jzp, kzp, jtr, ktr, vij, vjk, vik, &
    !$omp& r2ij, r2jk, r2ik, rij, rjk, rik, c6ij, c6jk, c6ik, triple, &
    !$omp& r0ij, r0jk, r0ik, r0, r1, r2, r3, r5, rr, fdmp, ang, c9, dE, &
-   !$omp& swij, swjk, swik, dswdr, sw) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& swij, swjk, swik, dswdr, sw)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -190,20 +183,15 @@ subroutine get_atm_dispersion_energy(mol, trans, cutoff, width, s9, rs9, alp, rv
                   rr = ang*fdmp

                   dE = rr * c9 * triple * sw / 3.0_wp
-                  energy_local(iat) = energy_local(iat) - dE
-                  energy_local(jat) = energy_local(jat) - dE
-                  energy_local(kat) = energy_local(kat) - dE
+                  energy(iat) = energy(iat) - dE
+                  energy(jat) = energy(jat) - dE
+                  energy(kat) = energy(kat) - dE
                end do
             end do
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_atm_dispersion_energy_)
-   energy(:) = energy(:) + energy_local(:)
-   !$omp end critical (get_atm_dispersion_energy_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_atm_dispersion_energy

@@ -262,30 +250,17 @@ subroutine get_atm_dispersion_derivs(mol, trans, cutoff, width, s9, rs9, alp, rv
    real(wp) :: alp3, r0r1alp3
    real(wp) :: swij, swjk, swik, dswijdr, dswjkdr, dswikdr, sw

-   ! Thread-private arrays for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)
-   real(wp), allocatable :: dEdcn_local(:)
-   real(wp), allocatable :: gradient_local(:, :)
-   real(wp), allocatable :: sigma_local(:, :)

    cutoff2 = cutoff*cutoff
    alp3 = alp / 3.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy, gradient, sigma, dEdcn) &
    !$omp shared(mol, trans, c6, s9, rs9, alp, alp3, rvdw, cutoff2, cutoff, width, dc6dcn) &
    !$omp private(iat, jat, kat, izp, jzp, kzp, jtr, ktr, ic, jc, vij, vjk, vik, &
    !$omp& r2ij, r2jk, r2ik, rij, rjk, rik, c6ij, c6jk, c6ik, triple, &
    !$omp& r0ij, r0jk, r0ik, r0, r1, r2, r3, r5, rr, fdmp, dfdmp, ang, &
    !$omp& dang, c9, dE, dE0, dGij, dGjk, dGik, dS, r0r1alp3, swij, &
-   !$omp& swjk, swik, dswijdr, dswjkdr, dswikdr, sw) &
-   !$omp shared(energy, gradient, sigma, dEdcn) &
-   !$omp private(energy_local, gradient_local, sigma_local, dEdcn_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   allocate(dEdcn_local(size(dEdcn, 1)), source=0.0_wp)
-   allocate(gradient_local(size(gradient, 1), size(gradient, 2)), source=0.0_wp)
-   allocate(sigma_local(size(sigma, 1), size(sigma, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& swjk, swik, dswijdr, dswjkdr, dswikdr, sw)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -359,13 +334,13 @@ subroutine get_atm_dispersion_derivs(mol, trans, cutoff, width, s9, rs9, alp, rv
                      & - dE0 * dswjkdr / rjk * swij * swik * vjk

                   dE = dE0 * triple * sw
-                  energy_local(iat) = energy_local(iat) - dE/3.0_wp
-                  energy_local(jat) = energy_local(jat) - dE/3.0_wp
-                  energy_local(kat) = energy_local(kat) - dE/3.0_wp
+                  energy(iat) = energy(iat) - dE/3.0_wp
+                  energy(jat) = energy(jat) - dE/3.0_wp
+                  energy(kat) = energy(kat) - dE/3.0_wp

-                  gradient_local(:, iat) = gradient_local(:, iat) - (dGij + dGik) * triple
-                  gradient_local(:, jat) = gradient_local(:, jat) + (dGij - dGjk) * triple
-                  gradient_local(:, kat) = gradient_local(:, kat) + (dGik + dGjk) * triple
+                  gradient(:, iat) = gradient(:, iat) - (dGij + dGik) * triple
+                  gradient(:, jat) = gradient(:, jat) + (dGij - dGjk) * triple
+                  gradient(:, kat) = gradient(:, kat) + (dGik + dGjk) * triple

                   do ic = 1, 3
                      do jc = 1, 3
@@ -374,31 +349,20 @@ subroutine get_atm_dispersion_derivs(mol, trans, cutoff, width, s9, rs9, alp, rv
                      end do
                   end do

-                  sigma_local(:, :) = sigma_local + dS * triple
+                  sigma(:, :) = sigma(:, :) + dS * triple

-                  dEdcn_local(iat) = dEdcn_local(iat) - dE * 0.5_wp &
+                  dEdcn(iat) = dEdcn(iat) - dE * 0.5_wp &
                      & * (dc6dcn(iat, jat) / c6ij + dc6dcn(iat, kat) / c6ik)
-                  dEdcn_local(jat) = dEdcn_local(jat) - dE * 0.5_wp &
+                  dEdcn(jat) = dEdcn(jat) - dE * 0.5_wp &
                      & * (dc6dcn(jat, iat) / c6ij + dc6dcn(jat, kat) / c6jk)
-                  dEdcn_local(kat) = dEdcn_local(kat) - dE * 0.5_wp &
+                  dEdcn(kat) = dEdcn(kat) - dE * 0.5_wp &
                      & * (dc6dcn(kat, iat) / c6ik + dc6dcn(kat, jat) / c6jk)
                end do
             end do
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_atm_dispersion_derivs_)
-   energy(:) = energy(:) + energy_local(:)
-   dEdcn(:) = dEdcn(:) + dEdcn_local(:)
-   gradient(:, :) = gradient(:, :) + gradient_local(:, :)
-   sigma(:, :) = sigma(:, :) + sigma_local(:, :)
-   !$omp end critical (get_atm_dispersion_derivs_)
-   deallocate(energy_local)
-   deallocate(dEdcn_local)
-   deallocate(gradient_local)
-   deallocate(sigma_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_atm_dispersion_derivs

@@ -444,24 +408,17 @@ subroutine get_atm_pairwise_dispersion(mol, trans, cutoff, width, s9, rs9, alp,
    real(wp) :: cutoff2, c9, dE, alp3
    real(wp) :: swij, swjk, swik, dswdr, sw

-   ! Thread-private arrays for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:, :)

    if (abs(s9) < epsilon(1.0_wp)) return
    cutoff2 = cutoff*cutoff
    alp3 = alp / 3.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, trans, c6, cutoff2, cutoff, width, s9, rs9, alp3, rvdw) &
    !$omp private(iat, jat, kat, izp, jzp, kzp, jtr, ktr, vij, vjk, vik, &
    !$omp& r2ij, r2jk, r2ik, rij, rjk, rik, c6ij, c6jk, c6ik, triple, &
    !$omp& r0ij, r0jk, r0ik, r0, r1, r2, r3, r5, rr, fdmp, ang, c9, dE, &
-   !$omp& swij, swjk, swik, dswdr, sw) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1), size(energy, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& swij, swjk, swik, dswdr, sw)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -507,23 +464,18 @@ subroutine get_atm_pairwise_dispersion(mol, trans, cutoff, width, s9, rs9, alp,
                   rr = ang*fdmp

                   dE = rr * c9 * triple * sw / 6.0_wp
-                  energy_local(jat, iat) = energy_local(jat, iat) - dE
-                  energy_local(kat, iat) = energy_local(kat, iat) - dE
-                  energy_local(iat, jat) = energy_local(iat, jat) - dE
-                  energy_local(kat, jat) = energy_local(kat, jat) - dE
-                  energy_local(iat, kat) = energy_local(iat, kat) - dE
-                  energy_local(jat, kat) = energy_local(jat, kat) - dE
+                  energy(jat, iat) = energy(jat, iat) - dE
+                  energy(kat, iat) = energy(kat, iat) - dE
+                  energy(iat, jat) = energy(iat, jat) - dE
+                  energy(kat, jat) = energy(kat, jat) - dE
+                  energy(iat, kat) = energy(iat, kat) - dE
+                  energy(jat, kat) = energy(jat, kat) - dE
                end do
             end do
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_atm_pairwise_dispersion_)
-   energy(:, :) = energy(:, :) + energy_local(:, :)
-   !$omp end critical (get_atm_pairwise_dispersion_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_atm_pairwise_dispersion

diff --git a/src/dftd3/damping/cso.f90 b/src/dftd3/damping/cso.f90
index 4253577..8fc4032 100644
--- a/src/dftd3/damping/cso.f90
+++ b/src/dftd3/damping/cso.f90
@@ -172,20 +172,13 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: c6ij, edisp, dE, sw, dswdr
    real(wp) :: d6_6, a2r0ij, r4

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, rrij, r0ij, rij, d6, ef, &
-   !$omp& sf, sig, t6, c6ij, edisp, dE, d6_6, a2r0ij, r4, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& sf, sig, t6, c6ij, edisp, dE, d6_6, a2r0ij, r4, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -213,19 +206,14 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(iat) = energy_local(iat) + dE
+            energy(iat) = energy(iat) + dE
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
+               energy(jat) = energy(jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_energy_cso_)
-   energy(:) = energy(:) + energy_local(:)
-   !$omp end critical (get_dispersion_energy_cso_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_energy

@@ -279,27 +267,14 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: dE, dG(3), dS(3, 3)
    real(wp) :: d6_6, a2r0ij, r4

-   ! Thread-private arrays for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)
-   real(wp), allocatable :: dEdcn_local(:)
-   real(wp), allocatable :: gradient_local(:, :)
-   real(wp), allocatable :: sigma_local(:, :)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy, gradient, sigma, dEdcn) &
    !$omp shared(mol, self, c6, dc6dcn, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, ic, jc, vec, r2, rrij, r0ij, rij, d6, ef, &
    !$omp& sf, sig, t6, c6ij, dsig, dt6, edisp0, gdisp0, edisp, gdisp, dE, &
-   !$omp& dG, dS, d6_6, a2r0ij, r4, sw, dswdr) &
-   !$omp shared(energy, gradient, sigma, dEdcn) &
-   !$omp private(energy_local, gradient_local, sigma_local, dEdcn_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   allocate(dEdcn_local(size(dEdcn, 1)), source=0.0_wp)
-   allocate(gradient_local(size(gradient, 1), size(gradient, 2)), source=0.0_wp)
-   allocate(sigma_local(size(sigma, 1), size(sigma, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& dG, dS, d6_6, a2r0ij, r4, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -339,31 +314,20 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
                end do
             end do

-            energy_local(iat) = energy_local(iat) + dE
-            dEdcn_local(iat) = dEdcn_local(iat) - dc6dcn(iat, jat) * edisp
-            sigma_local(:, :) = sigma_local(:, :) + dS
+            energy(iat) = energy(iat) + dE
+            dEdcn(iat) = dEdcn(iat) - dc6dcn(iat, jat) * edisp
+            sigma(:, :) = sigma(:, :) + dS
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
-               dEdcn_local(jat) = dEdcn_local(jat) - dc6dcn(jat, iat) * edisp
-               gradient_local(:, iat) = gradient_local(:, iat) + dG
-               gradient_local(:, jat) = gradient_local(:, jat) - dG
-               sigma_local(:, :) = sigma_local(:, :) + dS
+               energy(jat) = energy(jat) + dE
+               dEdcn(jat) = dEdcn(jat) - dc6dcn(jat, iat) * edisp
+               gradient(:, iat) = gradient(:, iat) + dG
+               gradient(:, jat) = gradient(:, jat) - dG
+               sigma(:, :) = sigma(:, :) + dS
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_derivs_cso_)
-   energy(:) = energy(:) + energy_local(:)
-   dEdcn(:) = dEdcn(:) + dEdcn_local(:)
-   gradient(:, :) = gradient(:, :) + gradient_local(:, :)
-   sigma(:, :) = sigma(:, :) + sigma_local(:, :)
-   !$omp end critical (get_dispersion_derivs_cso_)
-   deallocate(energy_local)
-   deallocate(dEdcn_local)
-   deallocate(gradient_local)
-   deallocate(sigma_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_derivs

@@ -452,20 +416,13 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,
    real(wp) :: c6ij, edisp, dE, sw, dswdr
    real(wp) :: d6_6, a2r0ij, r4

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:, :)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, rrij, r0ij, rij, d6, ef, &
-   !$omp& sf, sig, t6, c6ij, edisp, dE, d6_6, a2r0ij, r4, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1), size(energy, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& sf, sig, t6, c6ij, edisp, dE, d6_6, a2r0ij, r4, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -493,19 +450,14 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(jat, iat) = energy_local(jat, iat) + dE
+            energy(jat, iat) = energy(jat, iat) + dE
             if (iat /= jat) then
-               energy_local(iat, jat) = energy_local(iat, jat) + dE
+               energy(iat, jat) = energy(iat, jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_pairwise_dispersion2_cso_)
-   energy(:, :) = energy(:, :) + energy_local(:, :)
-   !$omp end critical (get_pairwise_dispersion2_cso_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_pairwise_dispersion2

diff --git a/src/dftd3/damping/mzero.f90 b/src/dftd3/damping/mzero.f90
index 301aaf1..230c965 100644
--- a/src/dftd3/damping/mzero.f90
+++ b/src/dftd3/damping/mzero.f90
@@ -172,22 +172,15 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: edisp, cutoff2, r0ij, rrij, c6ij, dE
    real(wp) :: irs6r0, irs8r0, betr0, sw, dswdr

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)

    cutoff2 = cutoff*cutoff
    alp6 = self%alp
    alp8 = self%alp + 2.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, alp6, alp8, rvdw, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, r1, r6, r8, t6, t8, f6, &
-   !$omp& f8, edisp, r0ij, rrij, c6ij, dE, irs6r0, irs8r0, betr0, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& f8, edisp, r0ij, rrij, c6ij, dE, irs6r0, irs8r0, betr0, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -218,19 +211,14 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(iat) = energy_local(iat) + dE
+            energy(iat) = energy(iat) + dE
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
+               energy(jat) = energy(jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_energy_)
-   energy(:) = energy(:) + energy_local(:)
-   !$omp end critical (get_dispersion_energy_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_energy

@@ -283,29 +271,16 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: edisp0, gdisp0, edisp, gdisp, cutoff2, r0ij, rrij, c6ij, dE, dG(3), dS(3, 3)
    real(wp) :: irs6r0, irs8r0, betr0, betr02rs6, betr02rs8, sw, dswdr

-   ! Thread-private arrays for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)
-   real(wp), allocatable :: dEdcn_local(:)
-   real(wp), allocatable :: gradient_local(:, :)
-   real(wp), allocatable :: sigma_local(:, :)

    cutoff2 = cutoff*cutoff
    alp6 = self%alp
    alp8 = self%alp + 2.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy, gradient, sigma, dEdcn) &
    !$omp shared(mol, self, c6, dc6dcn, trans, cutoff2, cutoff, width, alp6, alp8, rvdw, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, ic, jc, vec, r2, r1, r6, r8, t6, t8, d6, &
    !$omp& d8, f6, f8, edisp0, gdisp0, edisp, gdisp, r0ij, rrij, c6ij, &
-   !$omp& dE, dG, dS, irs6r0, irs8r0, betr0, betr02rs6, betr02rs8, sw, dswdr) &
-   !$omp shared(energy, gradient, sigma, dEdcn) &
-   !$omp private(energy_local, gradient_local, sigma_local, dEdcn_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   allocate(dEdcn_local(size(dEdcn, 1)), source=0.0_wp)
-   allocate(gradient_local(size(gradient, 1), size(gradient, 2)), source=0.0_wp)
-   allocate(sigma_local(size(sigma, 1), size(sigma, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& dE, dG, dS, irs6r0, irs8r0, betr0, betr02rs6, betr02rs8, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -352,31 +327,20 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
                end do
             end do

-            energy_local(iat) = energy_local(iat) + dE
-            dEdcn_local(iat) = dEdcn_local(iat) - dc6dcn(iat, jat) * edisp
-            sigma_local(:, :) = sigma_local + dS
+            energy(iat) = energy(iat) + dE
+            dEdcn(iat) = dEdcn(iat) - dc6dcn(iat, jat) * edisp
+            sigma(:, :) = sigma(:, :) + dS
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
-               dEdcn_local(jat) = dEdcn_local(jat) - dc6dcn(jat, iat) * edisp
-               gradient_local(:, iat) = gradient_local(:, iat) + dG
-               gradient_local(:, jat) = gradient_local(:, jat) - dG
-               sigma_local(:, :) = sigma_local + dS
+               energy(jat) = energy(jat) + dE
+               dEdcn(jat) = dEdcn(jat) - dc6dcn(jat, iat) * edisp
+               gradient(:, iat) = gradient(:, iat) + dG
+               gradient(:, jat) = gradient(:, jat) - dG
+               sigma(:, :) = sigma(:, :) + dS
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_derivs_)
-   energy(:) = energy(:) + energy_local(:)
-   dEdcn(:) = dEdcn(:) + dEdcn_local(:)
-   gradient(:, :) = gradient(:, :) + gradient_local(:, :)
-   sigma(:, :) = sigma(:, :) + sigma_local(:, :)
-   !$omp end critical (get_dispersion_derivs_)
-   deallocate(energy_local)
-   deallocate(dEdcn_local)
-   deallocate(gradient_local)
-   deallocate(sigma_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_derivs

@@ -464,22 +428,15 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,
    real(wp) :: vec(3), r2, r1, r6, r8, t6, t8, f6, f8, alp6, alp8
    real(wp) :: edisp, cutoff2, r0ij, rrij, c6ij, dE, sw, dswdr

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:, :)

    cutoff2 = cutoff*cutoff
    alp6 = self%alp
    alp8 = self%alp + 2.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, alp6, alp8, rvdw, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, r1, r6, r8, t6, t8, f6, &
-   !$omp& f8, edisp, r0ij, rrij, c6ij, dE, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1), size(energy, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& f8, edisp, r0ij, rrij, c6ij, dE, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -507,19 +464,14 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(jat, iat) = energy_local(jat, iat) + dE
+            energy(jat, iat) = energy(jat, iat) + dE
             if (iat /= jat) then
-               energy_local(iat, jat) = energy_local(iat, jat) + dE
+               energy(iat, jat) = energy(iat, jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_pairwise_dispersion2_)
-   energy(:, :) = energy(:, :) + energy_local(:, :)
-   !$omp end critical (get_pairwise_dispersion2_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_pairwise_dispersion2

diff --git a/src/dftd3/damping/optimizedpower.f90 b/src/dftd3/damping/optimizedpower.f90
index 9f365ac..0def41c 100644
--- a/src/dftd3/damping/optimizedpower.f90
+++ b/src/dftd3/damping/optimizedpower.f90
@@ -172,21 +172,14 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: r0ij2, r0ij6, r0ij8, abr0ij6, abr0ij8, r4
    real(wp) :: sw, dswdr

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, r1, r0ij, rrij, c6ij, t6, &
    !$omp& t8, edisp, dE, rb, ab, r0ij2, r0ij6, r0ij8, abr0ij6, abr0ij8, &
-   !$omp& r4, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& r4, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -216,19 +209,14 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(iat) = energy_local(iat) + dE
+            energy(iat) = energy(iat) + dE
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
+               energy(jat) = energy(jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_energy_)
-   energy(:) = energy(:) + energy_local(:)
-   !$omp end critical (get_dispersion_energy_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_energy

@@ -282,27 +270,14 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: dE, dG(3), dS(3, 3), rb, ab
    real(wp) :: r0ij2, r0ij6, r0ij8, abr0ij6, abr0ij8, r4

-   ! Thread-private arrays for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)
-   real(wp), allocatable :: dEdcn_local(:)
-   real(wp), allocatable :: gradient_local(:, :)
-   real(wp), allocatable :: sigma_local(:, :)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy, gradient, sigma, dEdcn) &
    !$omp shared(mol, self, c6, dc6dcn, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, ic, jc, vec, r2, r1, r0ij, rrij, c6ij, &
    !$omp& t6, t8, d6, d8, edisp0, gdisp0, edisp, gdisp, dE, dG, dS, rb, ab, &
-   !$omp& r0ij2, r0ij6, r0ij8, abr0ij6, abr0ij8, r4, sw, dswdr) &
-   !$omp shared(energy, gradient, sigma, dEdcn) &
-   !$omp private(energy_local, gradient_local, sigma_local, dEdcn_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   allocate(dEdcn_local(size(dEdcn, 1)), source=0.0_wp)
-   allocate(gradient_local(size(gradient, 1), size(gradient, 2)), source=0.0_wp)
-   allocate(sigma_local(size(sigma, 1), size(sigma, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& r0ij2, r0ij6, r0ij8, abr0ij6, abr0ij8, r4, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -344,31 +319,20 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
                end do
             end do

-            energy_local(iat) = energy_local(iat) + dE
-            dEdcn_local(iat) = dEdcn_local(iat) - dc6dcn(iat, jat) * edisp
-            sigma_local(:, :) = sigma_local + dS
+            energy(iat) = energy(iat) + dE
+            dEdcn(iat) = dEdcn(iat) - dc6dcn(iat, jat) * edisp
+            sigma(:, :) = sigma(:, :) + dS
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
-               dEdcn_local(jat) = dEdcn_local(jat) - dc6dcn(jat, iat) * edisp
-               gradient_local(:, iat) = gradient_local(:, iat) + dG
-               gradient_local(:, jat) = gradient_local(:, jat) - dG
-               sigma_local(:, :) = sigma_local + dS
+               energy(jat) = energy(jat) + dE
+               dEdcn(jat) = dEdcn(jat) - dc6dcn(jat, iat) * edisp
+               gradient(:, iat) = gradient(:, iat) + dG
+               gradient(:, jat) = gradient(:, jat) - dG
+               sigma(:, :) = sigma(:, :) + dS
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_derivs_)
-   energy(:) = energy(:) + energy_local(:)
-   dEdcn(:) = dEdcn(:) + dEdcn_local(:)
-   gradient(:, :) = gradient(:, :) + gradient_local(:, :)
-   sigma(:, :) = sigma(:, :) + sigma_local(:, :)
-   !$omp end critical (get_dispersion_derivs_)
-   deallocate(energy_local)
-   deallocate(dEdcn_local)
-   deallocate(gradient_local)
-   deallocate(sigma_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_derivs

@@ -456,20 +420,13 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,
    real(wp) :: vec(3), r2, r1, cutoff2, r0ij, rrij, c6ij, t6, t8, edisp, dE, rb, ab
    real(wp) :: sw, dswdr

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:, :)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, r1, r0ij, rrij, c6ij, t6, &
-   !$omp& t8, edisp, dE, rb, ab, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1), size(energy, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& t8, edisp, dE, rb, ab, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -493,19 +450,14 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(jat, iat) = energy_local(jat, iat) + dE
+            energy(jat, iat) = energy(jat, iat) + dE
             if (iat /= jat) then
-               energy_local(iat, jat) = energy_local(iat, jat) + dE
+               energy(iat, jat) = energy(iat, jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_pairwise_dispersion2_)
-   energy(:, :) = energy(:, :) + energy_local(:, :)
-   !$omp end critical (get_pairwise_dispersion2_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_pairwise_dispersion2

diff --git a/src/dftd3/damping/rational.f90 b/src/dftd3/damping/rational.f90
index a8edb7c..32aac1e 100644
--- a/src/dftd3/damping/rational.f90
+++ b/src/dftd3/damping/rational.f90
@@ -170,20 +170,13 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: sw, dswdr
    real(wp) :: r0ij2, r0ij6, r0ij8, r4

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, r1, r0ij, rrij, c6ij, t6, &
-   !$omp& t8, edisp, dE, r0ij2, r0ij6, r0ij8, r4, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& t8, edisp, dE, r0ij2, r0ij6, r0ij8, r4, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -209,19 +202,14 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(iat) = energy_local(iat) + dE
+            energy(iat) = energy(iat) + dE
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
+               energy(jat) = energy(jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_energy_)
-   energy(:) = energy(:) + energy_local(:)
-   !$omp end critical (get_dispersion_energy_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_energy

@@ -275,27 +263,14 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: dE, dG(3), dS(3, 3)
    real(wp) :: r0ij2, r0ij6, r0ij8, r4

-   ! Thread-private arrays for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)
-   real(wp), allocatable :: dEdcn_local(:)
-   real(wp), allocatable :: gradient_local(:, :)
-   real(wp), allocatable :: sigma_local(:, :)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy, gradient, sigma, dEdcn) &
    !$omp shared(mol, self, c6, dc6dcn, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, ic, jc, vec, r2, r1, r0ij, rrij, c6ij, &
    !$omp& t6, t8, d6, d8, edisp0, gdisp0, edisp, gdisp, dE, dG, dS, &
-   !$omp& r0ij2, r0ij6, r0ij8, r4, sw, dswdr) &
-   !$omp shared(energy, gradient, sigma, dEdcn) &
-   !$omp private(energy_local, gradient_local, sigma_local, dEdcn_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   allocate(dEdcn_local(size(dEdcn, 1)), source=0.0_wp)
-   allocate(gradient_local(size(gradient, 1), size(gradient, 2)), source=0.0_wp)
-   allocate(sigma_local(size(sigma, 1), size(sigma, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& r0ij2, r0ij6, r0ij8, r4, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -333,31 +308,20 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
                end do
             end do

-            energy_local(iat) = energy_local(iat) + dE
-            dEdcn_local(iat) = dEdcn_local(iat) - dc6dcn(iat, jat) * edisp
-            sigma_local(:, :) = sigma_local + dS
+            energy(iat) = energy(iat) + dE
+            dEdcn(iat) = dEdcn(iat) - dc6dcn(iat, jat) * edisp
+            sigma(:, :) = sigma(:, :) + dS
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
-               dEdcn_local(jat) = dEdcn_local(jat) - dc6dcn(jat, iat) * edisp
-               gradient_local(:, iat) = gradient_local(:, iat) + dG
-               gradient_local(:, jat) = gradient_local(:, jat) - dG
-               sigma_local(:, :) = sigma_local + dS
+               energy(jat) = energy(jat) + dE
+               dEdcn(jat) = dEdcn(jat) - dc6dcn(jat, iat) * edisp
+               gradient(:, iat) = gradient(:, iat) + dG
+               gradient(:, jat) = gradient(:, jat) - dG
+               sigma(:, :) = sigma(:, :) + dS
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_derivs_)
-   energy(:) = energy(:) + energy_local(:)
-   dEdcn(:) = dEdcn(:) + dEdcn_local(:)
-   gradient(:, :) = gradient(:, :) + gradient_local(:, :)
-   sigma(:, :) = sigma(:, :) + sigma_local(:, :)
-   !$omp end critical (get_dispersion_derivs_)
-   deallocate(energy_local)
-   deallocate(dEdcn_local)
-   deallocate(gradient_local)
-   deallocate(sigma_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_derivs

@@ -445,20 +409,13 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,
    real(wp) :: vec(3), r2, r1, cutoff2, r0ij, rrij, c6ij, t6, t8, edisp, dE
    real(wp) :: sw, dswdr

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:, :)

    cutoff2 = cutoff*cutoff

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, r1, r0ij, rrij, c6ij, t6, &
-   !$omp& t8, edisp, dE, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1), size(energy, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& t8, edisp, dE, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -480,19 +437,14 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(jat, iat) = energy_local(jat, iat) + dE
+            energy(jat, iat) = energy(jat, iat) + dE
             if (iat /= jat) then
-               energy_local(iat, jat) = energy_local(iat, jat) + dE
+               energy(iat, jat) = energy(iat, jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_pairwise_dispersion2_)
-   energy(:, :) = energy(:, :) + energy_local(:, :)
-   !$omp end critical (get_pairwise_dispersion2_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_pairwise_dispersion2

diff --git a/src/dftd3/damping/zero.f90 b/src/dftd3/damping/zero.f90
index 2e3ed6e..e5b78ad 100644
--- a/src/dftd3/damping/zero.f90
+++ b/src/dftd3/damping/zero.f90
@@ -170,22 +170,15 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: edisp, cutoff2, r0ij, rrij, c6ij, dE
    real(wp) :: rs6r0, rs8r0, sw, dswdr

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)

    cutoff2 = cutoff*cutoff
    alp6 = self%alp
    alp8 = self%alp + 2.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, alp6, alp8, rvdw, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, r1, r6, r8, t6, t8, f6, &
-   !$omp& f8, edisp, r0ij, rrij, c6ij, dE, rs6r0, rs8r0, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& f8, edisp, r0ij, rrij, c6ij, dE, rs6r0, rs8r0, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -215,19 +208,14 @@ subroutine get_dispersion_energy(self, mol, trans, cutoff, width, rvdw, r4r2, c6

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(iat) = energy_local(iat) + dE
+            energy(iat) = energy(iat) + dE
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
+               energy(jat) = energy(jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_energy_)
-   energy(:) = energy(:) + energy_local(:)
-   !$omp end critical (get_dispersion_energy_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_energy

@@ -280,29 +268,16 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
    real(wp) :: edisp0, gdisp0, edisp, gdisp, cutoff2, r0ij, rrij, c6ij, dE, dG(3), dS(3, 3)
    real(wp) :: rs6r0, rs8r0, sw, dswdr

-   ! Thread-private arrays for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:)
-   real(wp), allocatable :: dEdcn_local(:)
-   real(wp), allocatable :: gradient_local(:, :)
-   real(wp), allocatable :: sigma_local(:, :)

    cutoff2 = cutoff*cutoff
    alp6 = self%alp
    alp8 = self%alp + 2.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy, gradient, sigma, dEdcn) &
    !$omp shared(mol, self, c6, dc6dcn, trans, cutoff2, cutoff, width, alp6, alp8, r4r2, rvdw) &
    !$omp private(iat, jat, izp, jzp, jtr, ic, jc, vec, r2, r1, r6, r8, t6, t8, d6, &
    !$omp& d8, f6, f8, edisp0, gdisp0, edisp, gdisp, r0ij, rrij, c6ij, dE, &
-   !$omp& dG, dS, rs6r0, rs8r0, sw, dswdr) &
-   !$omp shared(energy, gradient, sigma, dEdcn) &
-   !$omp private(energy_local, gradient_local, sigma_local, dEdcn_local)
-   allocate(energy_local(size(energy, 1)), source=0.0_wp)
-   allocate(dEdcn_local(size(dEdcn, 1)), source=0.0_wp)
-   allocate(gradient_local(size(gradient, 1), size(gradient, 2)), source=0.0_wp)
-   allocate(sigma_local(size(sigma, 1), size(sigma, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& dG, dS, rs6r0, rs8r0, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -346,31 +321,20 @@ subroutine get_dispersion_derivs(self, mol, trans, cutoff, width, rvdw, r4r2, c6
                end do
             end do

-            energy_local(iat) = energy_local(iat) + dE
-            dEdcn_local(iat) = dEdcn_local(iat) - dc6dcn(iat, jat) * edisp
-            sigma_local(:, :) = sigma_local + dS
+            energy(iat) = energy(iat) + dE
+            dEdcn(iat) = dEdcn(iat) - dc6dcn(iat, jat) * edisp
+            sigma(:, :) = sigma(:, :) + dS
             if (iat /= jat) then
-               energy_local(jat) = energy_local(jat) + dE
-               dEdcn_local(jat) = dEdcn_local(jat) - dc6dcn(jat, iat) * edisp
-               gradient_local(:, iat) = gradient_local(:, iat) + dG
-               gradient_local(:, jat) = gradient_local(:, jat) - dG
-               sigma_local(:, :) = sigma_local + dS
+               energy(jat) = energy(jat) + dE
+               dEdcn(jat) = dEdcn(jat) - dc6dcn(jat, iat) * edisp
+               gradient(:, iat) = gradient(:, iat) + dG
+               gradient(:, jat) = gradient(:, jat) - dG
+               sigma(:, :) = sigma(:, :) + dS
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_dispersion_derivs_)
-   energy(:) = energy(:) + energy_local(:)
-   dEdcn(:) = dEdcn(:) + dEdcn_local(:)
-   gradient(:, :) = gradient(:, :) + gradient_local(:, :)
-   sigma(:, :) = sigma(:, :) + sigma_local(:, :)
-   !$omp end critical (get_dispersion_derivs_)
-   deallocate(energy_local)
-   deallocate(dEdcn_local)
-   deallocate(gradient_local)
-   deallocate(sigma_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_dispersion_derivs

@@ -459,22 +423,15 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,
    real(wp) :: edisp, cutoff2, r0ij, rrij, c6ij, dE
    real(wp) :: rs6r0, rs8r0, sw, dswdr

-   ! Thread-private array for reduction
-   ! Set to 0 explicitly as the shared variants are potentially non-zero (inout)
-   real(wp), allocatable :: energy_local(:, :)

    cutoff2 = cutoff*cutoff
    alp6 = self%alp
    alp8 = self%alp + 2.0_wp

-   !$omp parallel default(none) &
+   !$omp parallel do schedule(runtime) default(none) reduction(+:energy) &
    !$omp shared(mol, self, c6, trans, cutoff2, cutoff, width, alp6, alp8, rvdw, r4r2) &
    !$omp private(iat, jat, izp, jzp, jtr, vec, r2, r1, r6, r8, t6, t8, f6, &
-   !$omp& f8, edisp, r0ij, rrij, c6ij, dE, rs6r0, rs8r0, sw, dswdr) &
-   !$omp shared(energy) &
-   !$omp private(energy_local)
-   allocate(energy_local(size(energy, 1), size(energy, 2)), source=0.0_wp)
-   !$omp do schedule(runtime)
+   !$omp& f8, edisp, r0ij, rrij, c6ij, dE, rs6r0, rs8r0, sw, dswdr)
    do iat = 1, mol%nat
       izp = mol%id(iat)
       do jat = 1, iat
@@ -504,19 +461,14 @@ subroutine get_pairwise_dispersion2(self, mol, trans, cutoff, width, rvdw, r4r2,

             dE = -c6ij*edisp * 0.5_wp

-            energy_local(jat, iat) = energy_local(jat, iat) + dE
+            energy(jat, iat) = energy(jat, iat) + dE
             if (iat /= jat) then
-               energy_local(iat, jat) = energy_local(iat, jat) + dE
+               energy(iat, jat) = energy(iat, jat) + dE
             end if
          end do
       end do
    end do
-   !$omp end do
-   !$omp critical (get_pairwise_dispersion2_)
-   energy(:, :) = energy(:, :) + energy_local(:, :)
-   !$omp end critical (get_pairwise_dispersion2_)
-   deallocate(energy_local)
-   !$omp end parallel
+   !$omp end parallel do

 end subroutine get_pairwise_dispersion2
